### PR TITLE
Make adapters config work more natively

### PR DESCRIPTION
**Pull request by dsalahutdinov**
A little bit clearly code of adapters config

### CODE REVIEW

**Comment by dsalahutdinov** [comment]
Supposing profit of this PR is:
 1) less code, less problems :), especially when you have method_missing 😄 
 2) it fails more clearby clearly with NoMethodError, in case you fogot to require some lib (and the adapter was not plugged in)

**Comment by palkan** [comment]
Having `adapters` behave like a Hash has some advantages: we can list (iterate) adapters, delete and re-register them.

> it fails more clearly with NoMethodError,

That makes sense. Maybe, it's better to make `adapters` a wrapper over Hash instance and delegate Enumerable methods to it (i.e. `include Enumerable` and define `each` method).

> it works much faster 

It doesn't matter in our case)

**Comment by dsalahutdinov** [comment]
Made some changes about wrapping the hash, does it now look better?

File: lib/isolator/adapters/config.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+require "forwardable"
+
+module Isolator
+  module Adapters
+    # Wraps the adapter config and provides accessor by name
+    class Config
+      include Enumerable
+      extend Forwardable
+      def_delegators :hash_config, :each, :[]
```
Comment (palkan) [inline lib/isolator/adapters/config.rb]: Let's use a shorter name, e.g. `store`

File: lib/isolator/adapters/config.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+      def register(id, adapter)
+        raise "Adapter already registered: #{id}" if respond_to?(id)
+        define_singleton_method(id) { self[id] }
```
Comment (palkan) [inline lib/isolator/adapters/config.rb]: Let's use the  previous `method_missing` approach but without fallback to `super` (in order to raise `NoMethodError` when no such key).

The provided approach doesn't play well with deleting keys: after a key has been deleted we except to got an exception; and now it returns `nil` instead.

**Comment by dsalahutdinov** [comment]
I've changed code for use `method_missing`

File: lib/isolator/isolate.rb [hunk 3]
```diff
@@ -4,9 +4,9 @@ module Isolator
   # Add .isolate function to build and register adapters
   module Isolate
     def isolate(id, **options)
-      raise "Adapter already registered: #{id}" if Isolator.adapters.key?(id.to_s)
-      adapter = AdapterBuilder.call(**options)
-      Isolator.adapters[id.to_s] = adapter
+      AdapterBuilder.call(**options).tap do |adapter|
```
Comment (palkan) [inline lib/isolator/isolate.rb]: We should raise error before `.call`, 'cause we're patching a target module in `.call`.

**Comment by palkan** [comment]
Let's leave everything as is; no need for complexity

**Comment by dsalahutdinov** [comment]
:+1: yes, it became too complicated. Thank you.

**Pull request closed without merging**